src/app.tsx

```diff
@@ -326,20 +335,22 @@ export default function Chat() {
                   ) {
                     e.preventDefault();
                     handleAgentSubmit(e as unknown as React.FormEvent);
+                    setTextareaHeight("auto"); // Reset height on Enter submission
                   }
                 }}
-                onValueChange={undefined}
+                rows={2}
+                style={{ height: textareaHeight }}
               />
+              <div className="absolute bottom-0 right-0 p-2 w-fit flex flex-row justify-end">
+                <button
+                  type="submit"
+                  className="inline-flex items-center cursor-pointer justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground hover:bg-primary/90 rounded-full p-1.5 h-fit border dark:border-zinc-600"
```

Comment:
spotted `dark:border-zinc-600`, for borders I preferred `border-neutral-200 dark:border-neutral-800` throughout